app/models/note.rb

```diff
@@ -4,8 +4,21 @@ class Note < ApplicationRecord
   validates :body, presence: true
   belongs_to :submission
 
-  belongs_to :author,
-             foreign_key: :gravity_user_id,
-             class_name: 'User',
-             primary_key: :gravity_user_id
+  attr_reader :author
+
+  after_initialize :set_author
```

Comment:
It's pretty unconventional to request external data just to initialize a model object. Initializing model objects happens throughout tests and production operation where a complete Gravity response might not be necessary. Instead, I suggest adding a helper method (like `get_author`) to the model that lazily loads and memoizes the external data.